packages/@uppy/aws-s3-multipart/src/createSignedURL.js

```diff
@@ -76,6 +76,30 @@ async function hash (key, data) {
   return subtle.sign(algorithm, await generateHmacKey(key), ec.encode(data))
 }
 
+/**
+ * @see
+ *   https://github.com/smithy-lang/smithy-typescript/blob/main/packages/smithy-client/src/extended-encode-uri-component.ts
+ */
+function extendedEncodeURIComponent(str) {
+  return encodeURIComponent(str).replace(/[!'()*]/g, (c) =>
+    `%${c.charCodeAt(0).toString(16).toUpperCase()}`
+  )
+}
+
+class Query {
+  #params = []
+
+  append(key, value) {
+    this.#params.push([key, value])
+  }
+
+  toString() {
+    return this.#params
+      .map(([key, value]) => `${extendedEncodeURIComponent(key)}=${extendedEncodeURIComponent(value)}`)
+      .join('&')
+  }
+}
+
```

Comment:
I'm pretty sure we don't need that, the browser will already escape all the characters – escape `.` and `*`, but I don't think those are problematic.
```suggestion
```

Reply:
It means, The query parameters within signed URL does not have `.*` so not escaping such 2 chars is NOT problem, right?

Reply:
If above question is right, you can confirm with fix:
https://github.com/transloadit/uppy/pull/5006/commits/3ab6024bfca48658a7fea4a4e202c3033a22d5a8